src/overlays/actors/ovl_Obj_Hariko/z_obj_hariko.c

```diff
@@ -31,24 +33,70 @@ const ActorInit Obj_Hariko_InitVars = {
     (ActorFunc)ObjHariko_Draw,
 };
 
-#endif
-
-extern UNK_TYPE D_06000080;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Hariko/ObjHariko_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Hariko/ObjHariko_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Hariko/func_80B66A7C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Hariko/func_80B66A90.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Hariko/func_80B66AA0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Hariko/func_80B66AC4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Hariko/func_80B66B78.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Hariko/ObjHariko_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Hariko/ObjHariko_Draw.s")
+void ObjHariko_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjHariko* this = THIS;
+    Actor_SetScale(&this->actor, 0.1f);
+    this->headRotation.x = 0;
+    this->headRotation.y = 0;
+    this->headRotation.z = 0;
+    this->headOffset = 0;
+    this->bobbleStep = 0.0f;
+    ObjHariko_SetNoAction(this);
+}
+
+void ObjHariko_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void ObjHariko_SetNoAction(ObjHariko* this) {
+    this->actionFunc = ObjHariko_NoAction;
+}
+
+void ObjHariko_NoAction(ObjHariko* this, GlobalContext* globalCtx) {
+}
+
+void ObjHariko_StartBobbing(Actor* thisx) {
+    ObjHariko* this = THIS;
+    this->bobbleStep = 2730.0f;
+    this->unk154 = 0;
+    this->actionFunc = ObjHariko_BobHead;
+}
+
+void ObjHariko_BobHead(ObjHariko* this, GlobalContext* globalCtx) {
+    this->headOffset = this->headOffset + 5461;
+    this->headRotation.x = Math_SinS(this->headOffset) * this->bobbleStep;
+    this->headRotation.y = Math_CosS(this->headOffset) * this->bobbleStep;
+    Math_SmoothStepToF(&this->bobbleStep, 0, 0.5f, 18.0f, 18.0f);
+    if (this->bobbleStep < 182.0f) {
+        ObjHariko_SetNoAction(this);
+    }
+}
+
+void ObjHariko_CheckForQuakes(Actor* thisx) {
+    if (Quake_NumActiveQuakes() != 0) {
+        ObjHariko_StartBobbing(thisx);
+    }
+}
+
+void ObjHariko_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjHariko* this = THIS;
+    this->actionFunc(this, globalCtx);
+    ObjHariko_CheckForQuakes(thisx);
+}
+
+void ObjHariko_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    ObjHariko* this = THIS;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+    func_8012C28C(globalCtx->state.gfxCtx);
+
+    Matrix_StatePush();
+    Matrix_InsertXRotation_s(this->headRotation.x, 1);
+    Matrix_RotateY(this->headRotation.y, 1);
```

Comment:
```suggestion
    Matrix_InsertXRotation_s(this->headRotation.x, MTXMODE_APPLY);
    Matrix_RotateY(this->headRotation.y, MTXMODE_APPLY);
```
This is a macro that can be substituted in